OpenRA.Mods.Common/Activities/CaptureActor.cs

```diff
@@ -19,15 +21,15 @@ public class CaptureActor : Enter
 		readonly Actor actor;
 		readonly Building building;
 		readonly Capturable capturable;
-		readonly CapturesInfo capturesInfo;
+		readonly Captures captures;
 		readonly Health health;
 
-		public CaptureActor(Actor self, Actor target)
+		public CaptureActor(Actor self, Actor target, Captures capture)
```

Comment:
I don't think that passing in a pre-selected `Captures` is the best way to go here.  If the target actor toggles between two different `Captures` (e.g. to add an extra capture type) that are both valid for the current captor then the capture order shouldn't be cancelled.

I think it would be better to follow the example from `CaptureProperties` here by querying and caching all of the `Captures` traits here, and then checking `if (captures.All(c => c.IsTraitDisabled)` in `Tick` and `var capturesInfo = captures.FirstOrDefault(c => !c.IsTraitDisabled)` in `OnInside`'s `FrameEndTask`.  This then minimizes the additional changes that will be needed when we add support for conditional `Capturable` traits.